src/scheduler/node_info.c

```diff
@@ -4971,22 +4979,20 @@ parse_execvnode(char *execvnode, server_info *sinfo)
 			invalid = 1;
 
 		if (!invalid) {
-			free(simplespec);
-			simplespec = parse_plus_spec_mt_safe(NULL, &tailptr, &ret);
-			if (!ret) {
-				if (simplespec != NULL) {
-					ret = parse_node_resc_r(simplespec, &node_name, &num_el, &nlkv, &kv);
-					if (ret < 0)
-						invalid = 1;
-				}
+			simplespec = parse_plus_spec_r(tailptr, &tailptr, &hp);
+			if (simplespec != NULL) {
+				int ret;
```

Comment:
Do you need to free kv and excvndup here?

Reply:
I don't need to free it inside the loop, I do it immediately outside of it.